packages/ember-views/tests/views/view/stream_test.js

```diff
@@ -31,6 +31,44 @@ QUnit.test("can return a stream that is notified of changes", function() {
   run(view, 'set', 'controller.name', 'Max');
 });
 
+QUnit.test("replacing itermediate stream chain value", function() {
+  expect(7);
+
+  view = EmberView.create({
+    foo: { bar: 'raytiley' }
+  });
+
+  var count = 0;
+
+  var stream = view.getStream('view.foo.bar');
+  stream.value();
+  stream.subscribe(function() {
+    stream.value();
+    count++;
+  });
+
+  // run
+  view.set('foo', { bar: 'mmun' });
```

Comment:
I think these might need to be wrapped in ember.run. Looks like `run` is used in other tests.